#!/usr/bin/env python

PACKAGE = "gr_pointcloud_filter"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("visualize_pointcloud",  bool_t,   0, "Filtered PointCloud Visualization ",  False)
gen.add("enable_filters",  bool_t,   0, "Enable Filters ",  True)
gen.add("voxel_filter",  bool_t,   0, "Enable Voxel Filter ",  False)
gen.add("radius_outlier_removal",  bool_t,   0, "Enable Radius Outlier Removal ",  True)
gen.add("conditional_filter",  bool_t,   0, "Enable Conditional Filter ",  True)
gen.add("ground_removal",  bool_t,   0, "Enable Ground Removal ",  True)
gen.add("outlier_removal",  bool_t,   0, "Enable Outlier Removal ",  True)

gen.add("leaf_size", double_t, 0 , "Voxel Cube Size", 0.3, 0,   5.0)
gen.add("min_points_per_voxel", int_t, 0 , "Minimum points per voxel",3, 0,50)

gen.add("conditional_distance", double_t, 0 , "Plus/Minus Condition Distance",25, 0.0, 200.0)

gen.add("max_iterations", int_t, 0, "Max Number of Iterations", 100, 0, 2000)
gen.add("distance_threshold", double_t, 0 , "Distance Threshold",0.3, 0, 1)
gen.add("eps_angle", double_t, 0 , "Eps Angle Tolerance ",10, 0, 30)

gen.add("optimize_coefficients", bool_t, 0 , "Optimize Coefficients", True)
gen.add("set_negative", bool_t, 0 , "Negative", True)
gen.add("mean_k", int_t, 0 , "Neighbors Number",15, 0, 200)
gen.add("std_threshold", double_t, 0 , "Standard Deviation Threshold",1.0, 0.0, 10)

gen.add("min_radius_removal", double_t, 0 , "Min Radius Threshold",1.0, 0.0, 10)
gen.add("min_neighbours", int_t, 0, "Min Number of Neighbours", 5, 0, 200)

gen.add("cluster_tolerance", double_t, 0, "Cluster Tolerance", 0.5,0.01, 20.0)
gen.add("min_cluster_neighbours", int_t, 0, "Min Number of Neighbours for Region Growing", 10, 0, 200)
gen.add("max_cluster_neighbours", int_t, 0, "Max Number of Neighbours for Region Growing", 40, 0, 200)

exit(gen.generate(PACKAGE, "gr_pointcloud_filter", "Filters"))
